src/mobile/src/components/Post/Pagination.jsx

```diff
@@ -0,0 +1,97 @@
+import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+import { MaterialIcons } from '@expo/vector-icons';
+
+const styles = StyleSheet.create({
+  pagination: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  paginationButton: {
+    padding: 12,
+    borderColor: '#9CA3AF85',
+    borderWidth: 1,
+    flex: 1,
+    margin: 12,
+    borderRadius: 8,
+    backgroundColor: '#FFFFFF',
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  paginationButtonLeft: {
+    justifyContent: 'flex-start',
+  },
+  paginationButtonRight: {
+    justifyContent: 'flex-end',
+  },
+  paginationButtonDisable: {
+    backgroundColor: '#FFFFFF50',
+    color: '#00000050',
+    borderColor: '#9CA3AF50',
+  },
+});
+
+const Pagination = ({
+  previousPostUrl,
+  previousPostUrlError,
+  nextPostUrl,
+  nextPostUrlError,
+  page,
+  navigation,
+}) => {
+  return (
+    <View style={styles.pagination}>
+      <TouchableOpacity
+        activeOpacity={0.5}
+        disabled={previousPostUrlError || !previousPostUrl ? true : false}
+        style={[
+          styles.paginationButton,
+          styles.paginationButtonLeft,
+          previousPostUrlError || !previousPostUrl ? styles.paginationButtonDisable : null,
+        ]}
+        onPress={() => {
+          navigation.replace('Post', { postURL: previousPostUrl[0].url, page: page - 1 });
+        }}
+      >
+        <>
+          <MaterialIcons
+            name="navigate-before"
+            size={24}
+            color={previousPostUrlError || !previousPostUrl ? '#00000050' : '#000000'}
+          />
+          <Text
+            style={previousPostUrlError || !previousPostUrl ? styles.paginationButtonDisable : null}
+          >
+            Previous Post
+          </Text>
+        </>
+      </TouchableOpacity>
```

Comment:
Could you extract this button into a reusable component?

Reply:
The `Next Post` and `Previous Post` buttons are a bit different.

Reply:
You could pass prop to it, like <NavButton title="next">

Reply:
Updated